src/core/trulens/core/tru.py

```diff
@@ -103,6 +104,9 @@ class Tru(python.SingletonPerName):
             on SQLAlchemy database URLs. (defaults to
             `sqlite://DEFAULT_DATABASE_FILE`).
 
+        database_engine: SQL-Alchemy compatible database engine.
```

Comment:
There are some things you can do (such as keypair auth) that you can't do with the URL alone.

Could also just make this a snowflake-specific thing, but thought it would be better to have this option.